pgmax/fg/graph.py

```diff
@@ -180,31 +187,68 @@ def _register_factor_group(
                 self._factor_group_to_msgs_starts[factor_group]
                 + factor_num_states_cumsum[vv]
             )
-            self._factor_to_potentials_starts[factor] = (
-                self._factor_group_to_potentials_starts[factor_group]
-                + vv * factor.log_potentials.shape[0]
-            )
-            factor_group_num_configs += factor.log_potentials.shape[0]
+
+            if isinstance(factor, nodes.EnumerationFactor):
+                self._factor_to_potentials_starts[factor] = (
+                    self._factor_group_to_potentials_starts[factor_group]
+                    + vv * factor.log_potentials.shape[0]
+                )
+                factor_group_num_configs += factor.log_potentials.shape[0]
 
         self._total_factor_num_states += factor_num_states_cumsum[-1]
-        self._total_factor_num_configs += factor_group_num_configs
         if name is not None:
             self._named_factor_groups[name] = factor_group
 
+        if isinstance(
+            factor_group, (groups.EnumerationFactorGroup, groups.PairwiseFactorGroup)
+        ):
+            self._total_factor_num_configs += factor_group_num_configs
+
     @cached_property
-    def wiring(self) -> nodes.EnumerationWiring:
+    def wiring(self) -> GraphWiring:
```

Comment:
This function should be updated in such a way that adding new factors should not involve updating this function.

A possible way of doing it:
- Organize all the factors according to types into a dictionary, which maps from factor types to the list of factors of that type.
- Inside this function, simply looping through the different factor types, call compile wiring on each factor, and call the appropriate concatenation function (probably also organized into a dictionary somewhere) to get the final wiring. The final wiring is also organized into a dictionary.

Additionally, we should minimize interdependency between different types of factors. Concretely, this means we assume factors are ordered according to their types (e.g. all the enumeration factors first, then all the OR factors), and the current `enum_factor_to_msgs_starts` and `or_factor_to_msgs_starts` should be replaced by single numbers (0 and a single offset respectively).

Reply:
We should discuss your last point. Currently factors and their dependencies (ie `_factor_to_msgs_starts` etc...) organized following the order in which they were added.
We can leave it this way. 
What you suggest is to add the factors then reorganize them, which may create some overhead.

Reply:
This is related to the comment for replacing the current tuple of factors with a dictionary. The only change would be to aggregate factors into two separate lists (according to their types) instead of one and I think there's minimal overhead